diff --git a/src/Ext/freecad/CMakeLists.txt b/src/Ext/freecad/CMakeLists.txt
index d9004275b6..baf6df2e52 100644
--- a/src/Ext/freecad/CMakeLists.txt
+++ b/src/Ext/freecad/CMakeLists.txt
@@ -1,5 +1,5 @@
 EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} -c 
-"from distutils.sysconfig import get_python_lib; print(get_python_lib())"
+"from distutils.sysconfig import get_python_lib; print(get_python_lib(prefix='', plat_specific=True))"
 OUTPUT_VARIABLE python_libs OUTPUT_STRIP_TRAILING_WHITESPACE )
 SET(PYTHON_MAIN_DIR ${python_libs})
 
