diff --git a/src/Mod/Drawing/Gui/DrawingView.cpp b/src/Mod/Drawing/Gui/DrawingView.cpp
index bbfb05465..5d06c061b 100644
--- a/src/Mod/Drawing/Gui/DrawingView.cpp
+++ b/src/Mod/Drawing/Gui/DrawingView.cpp
@@ -65,6 +65,11 @@
 
 using namespace DrawingGui;
 
+#ifdef B0
+// termios.h on macOS defines B0 which conflicts with QPrinter::B0
+#undef B0
+#endif
+
 SvgView::SvgView(QWidget *parent)
     : QGraphicsView(parent)
     , m_renderer(Native)
