diff --git a/src/Main/CMakeLists.txt b/src/Main/CMakeLists.txt
index 4948cb2..bebbc5e 100644
--- a/src/Main/CMakeLists.txt
+++ b/src/Main/CMakeLists.txt
@@ -49,7 +49,7 @@ if(BUILD_GUI)
             RUNTIME DESTINATION bin
             LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         )
-    elseif(APPLE)
+    elseif(APPLE AND NOT BUILD_WITH_CONDA)
         INSTALL(TARGETS FreeCADMain
 	    RUNTIME DESTINATION MacOS
             LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -94,7 +94,7 @@ if(WIN32)
         RUNTIME DESTINATION bin
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     )
-elseif(APPLE)
+elseif(APPLE AND NOT BUILD_WITH_CONDA)
     INSTALL(TARGETS FreeCADMainCmd
         RUNTIME DESTINATION MacOS
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
