From dd546a5ee99bbf565ee96b8e496c896e66670aa1 Mon Sep 17 00:00:00 2001
From: sprhawk <465558+sprhawk@users.noreply.github.com>
Date: Fri, 9 Jul 2021 19:22:08 +0800
Subject: [PATCH] to compile with PCL-1.12.0

---
 src/Mod/ReverseEngineering/App/SurfaceTriangulation.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Mod/ReverseEngineering/App/SurfaceTriangulation.cpp b/src/Mod/ReverseEngineering/App/SurfaceTriangulation.cpp
index 427fe176d4dc..6cafed46ddc9 100644
--- a/src/Mod/ReverseEngineering/App/SurfaceTriangulation.cpp
+++ b/src/Mod/ReverseEngineering/App/SurfaceTriangulation.cpp
@@ -48,6 +48,7 @@
 #include <pcl/surface/marching_cubes_hoppe.h>
 #include <pcl/surface/ear_clipping.h>
 #include <pcl/common/common.h>
+#include <pcl/common/io.h>
 #include <boost/random.hpp>
 #include <boost/math/special_functions/fpclassify.hpp>
 
