diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3a6b2c820..220f1bc37 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1022,39 +1022,39 @@ if(NOT FREECAD_LIBPACK_USE OR FREECAD_LIBPACK_CHECKFILE_CLBUNDLER)
         # CMake to fail to create Makefiles for a debug build.
         # So as a workaround we save and restore the value after checking for Shiboken2.
         set (SAVE_BUILD_TYPE ${CMAKE_BUILD_TYPE})
-        find_package(Shiboken2 QUIET)# REQUIRED
+        find_package(Shiboken2 REQUIRED)
         set (CMAKE_BUILD_TYPE ${SAVE_BUILD_TYPE})
-        if (Shiboken2_FOUND)
-            # Shiboken2 config file was found but it may use the wrong Python version
-            # Try to get the matching config suffix and repeat finding the package
-            if (PYTHON_VERSION_MAJOR LESS 3)
-                set(SHIBOKEN_PATTERN -python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR})
-            else()
-                set(SHIBOKEN_PATTERN .cpython-${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
-            endif()
-
-            file(GLOB SHIBOKEN_CONFIG "${Shiboken2_DIR}/Shiboken2Config${SHIBOKEN_PATTERN}*.cmake")
-            if (SHIBOKEN_CONFIG)
-                get_filename_component(SHIBOKEN_CONFIG_SUFFIX ${SHIBOKEN_CONFIG} NAME)
-                string(SUBSTRING ${SHIBOKEN_CONFIG_SUFFIX} 15 -1 SHIBOKEN_CONFIG_SUFFIX)
-                string(REPLACE ".cmake" "" PYTHON_CONFIG_SUFFIX ${SHIBOKEN_CONFIG_SUFFIX})
-                message(STATUS "PYTHON_CONFIG_SUFFIX: ${PYTHON_CONFIG_SUFFIX}")
-                find_package(Shiboken2 QUIET)
-            endif()
-        endif()
-
-        if(NOT SHIBOKEN_INCLUDE_DIR)
-            message("====================\n"
-                    "shiboken2 not found.\n"
-                    "====================\n")
-        endif(NOT SHIBOKEN_INCLUDE_DIR)
-
-        find_package(PySide2 QUIET)# REQUIRED
-        if(NOT PYSIDE_INCLUDE_DIR)
-            message("==================\n"
-                    "PySide2 not found.\n"
-                    "==================\n")
-        endif(NOT PYSIDE_INCLUDE_DIR)
+        #if (Shiboken2_FOUND)
+            ## Shiboken2 config file was found but it may use the wrong Python version
+            ## Try to get the matching config suffix and repeat finding the package
+            #if (PYTHON_VERSION_MAJOR LESS 3)
+                #set(SHIBOKEN_PATTERN -python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR})
+            #else()
+                #set(SHIBOKEN_PATTERN .cpython-${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
+            #endif()
+
+            #file(GLOB SHIBOKEN_CONFIG "${Shiboken2_DIR}/Shiboken2Config${SHIBOKEN_PATTERN}*.cmake")
+            #if (SHIBOKEN_CONFIG)
+                #get_filename_component(SHIBOKEN_CONFIG_SUFFIX ${SHIBOKEN_CONFIG} NAME)
+                #string(SUBSTRING ${SHIBOKEN_CONFIG_SUFFIX} 15 -1 SHIBOKEN_CONFIG_SUFFIX)
+                #string(REPLACE ".cmake" "" PYTHON_CONFIG_SUFFIX ${SHIBOKEN_CONFIG_SUFFIX})
+                #message(STATUS "PYTHON_CONFIG_SUFFIX: ${PYTHON_CONFIG_SUFFIX}")
+                #find_package(Shiboken2 QUIET)
+            #endif()
+        #endif()
+
+        #if(NOT SHIBOKEN_INCLUDE_DIR)
+            #message("====================\n"
+                    #"shiboken2 include dir not found.\n"
+                    #"====================\n")
+        #endif(NOT SHIBOKEN_INCLUDE_DIR)
+
+        find_package(PySide2 REQUIRED)
+        #if(NOT PYSIDE_INCLUDE_DIR)
+            #message("==================\n"
+                    #"PySide2 not found.\n"
+                    #"==================\n")
+        #endif(NOT PYSIDE_INCLUDE_DIR)
 
         find_package(PySide2Tools QUIET) #REQUIRED # PySide2 utilities (pyside2-uic & pyside2-rcc)
         if(NOT PYSIDE2_TOOLS_FOUND)
diff --git a/cMake/FindPySide2.cmake b/cMake/FindPySide2.cmake
new file mode 100644
index 000000000..d5975c822
--- /dev/null
+++ b/cMake/FindPySide2.cmake
@@ -0,0 +1,40 @@
+# - Find PySide2 installation
+#
+# The following variable are set:
+#   PySide2_FOUND      - PySide2 was found
+#   PYSIDE2_FOUND      - PySide2 was found
+#   PYSIDE2_LIBRARY    - The PySide2 libraries
+#   PYSIDE_INCLUDE_DIR - The PySide2 header-directory
+
+
+execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "
+import find_pyside2_helper
+print(find_pyside2_helper.find_pyside2_library())
+"
+    RESULT_VARIABLE FAILURE
+    OUTPUT_VARIABLE PYSIDE2_LIBRARY_PATH
+    WORKING_DIRECTORY CMAKE_CURRENT_SOURCE_DIR
+)
+
+execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "
+import find_pyside_helper
+print(find_pyside2_helper.find_pyside2_includes())
+"
+    OUTPUT_VARIABLE PYSIDE2_INCLUDE_PATH
+    WORKING_DIRECTORY CMAKE_CURRENT_SOURCE_DIR
+)
+
+set(PYSIDE2_FOUND FALSE)
+if(NOT FAILURE)
+    set(PYSIDE2_FOUND TRUE)
+    set(PYSIDE2_LIBRARY ${PYSIDE2_LIBRARY_PATH} CACHE PATH "The location of the PySide2 library")
+    set(PYSIDE2_INCLUDE_DIR ${PYSIDE2_INCLUDE_PATH} CACHE PATH "The location of PySide2 headers")
+endif(NOT FAILURE)
+
+if(NOT PYSIDE2_FOUND)
+    MESSAGE("==================\n"
+            "PySide2 not found.\n"
+            "==================\n")
+endif(NOT PYSIDE2_FOUND)
+
+set(PySide2_FOUND ${PYSIDE2_FOUND})
diff --git a/cMake/FindShiboken2.cmake b/cMake/FindShiboken2.cmake
new file mode 100644
index 000000000..e08b07700
--- /dev/null
+++ b/cMake/FindShiboken2.cmake
@@ -0,0 +1,42 @@
+# - Find Shiboken2 installation
+#
+# The following variable are set:
+#   Shiboken2_FOUND        - Shiboken2 was found
+#   Shiboken2_LIBRARY      - The Shiboken2 shared library
+#   Shiboken2_INCLUDE_DIR  - The Shiboken2 header directory
+
+
+execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} -c "
+import find_pyside2_helper
+print(find_pyside2_helper.find_shiboken2_library())
+"
+    RESULT_VARIABLE FAILURE
+    OUTPUT_VARIABLE SHIBOKEN_LIBRARY_PATH
+    WORKING_DIRECTORY CMAKE_CURRENT_SOURCE_DIR
+)
+
+execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} -c "
+import find_pyside2_helper
+print(find_pyside2_helper.find_shiboken2_library())
+"
+    RESULT_VARIABLE FAILURE
+    OUTPUT_VARIABLE SHIBOKEN_INCLUDE_PATH
+    WORKING_DIRECTORY CMAKE_CURRENT_SOURCE_DIR
+)
+
+set(SHIBOKEN2_FOUND FALSE)
+if(NOT FAILURE)
+    set(SHIBOKEN2_FOUND TRUE)
+    set(SHIBOKEN2_LIBRARY ${SHIBOKEN2_LIBRARY_PATH} CACHE PATH "The location of the SHIBOKEN2 library")
+    set(SHIBOKEN2_INCLUDE_DIR ${SHIBOKEN2_INCLUDE_PATH} CACHE PATH "The location of SHIBOKEN2 headers")
+endif(NOT FAILURE)
+
+if(NOT SHIBOKEN2_FOUND)
+    MESSAGE("==================\n"
+            "SHIBOKEN2 not found.\n"
+            "==================\n")
+endif(NOT SHIBOKEN2_FOUND)
+
+set(Shiboken2_FOUND ${SHIBOKEN2_FOUND})
diff --git a/cMake/find_pyside2_helper.py b/cMake/find_pyside2_helper.py
new file mode 100644
index 000000000..1ed0b7c68
--- /dev/null
+++ b/cMake/find_pyside2_helper.py
@@ -0,0 +1,70 @@
+import sysconfig
+import glob
+import os
+import sys
+
+info = sysconfig.get_paths()
+library_path = os.path.dirname(info["stdlib"])
+include_path = os.path.dirname(info["include"])
+
+DEBUGGING = False
+PYTHON_VERSION = str(sys.version_info.major) + str(sys.version_info.minor)
+
+LIB_SUFFIX = ".so"
+if sys.platform.startswith("win"):
+    LIB_SUFFIX = ".lib"
+elif sys.platform == "darwin":
+    LIB_SUFFIX = ".dylib"
+
+def debug_print(text):
+    if DEBUGGING:
+        print(text)
+
+def find_path(directory, test, target_name, find_dir=True):
+    """
+    returns all paths in directory where test returns True
+    test: function with args directory_path (root) and file name (fn)
+    """
+    found_files = []
+    for root, dirs, files in os.walk(directory):
+        for fn in files:
+            if test(root, fn):
+                if find_dir:
+                    found_files.append(root)
+                else:
+                    found_files.append(os.path.join(root, fn))
+    if len(found_files) == 0:
+        result = None
+        debug_print("{} not found in {}".format(target_name, directory))
+    elif len(found_files) > 0:
+        if len(found_files) > 1:
+            found_files.sort()
+            debug_print("{} paths possible for searchpath {}: {}".format(len(found_files), target_name, found_files))
+            debug_print("choosing first")
+        result = found_files[0]
+        debug_print("{} found in\n{}".format(target_name, result))
+    return result
+
+def pyside2_lib_test(root, fn):
+    return "pyside2" in fn and PYTHON_VERSION in fn and "packages" not in root and fn.endswith(LIB_SUFFIX)
+
+def pyside2_include_test(root, fn):
+    return "PySide2" in root and fn == "pyside.h"
+
+def shiboken2_lib_test(root, fn):
+    return "shiboken2" in fn and PYTHON_VERSION in fn and "packages" not in root and fn.endswith(LIB_SUFFIX)
+
+def shiboken2_include_test(root, fn):
+    return "shiboken2" in root and fn == "shiboken.h"
+
+def find_pyside2_library():
+    return find_path(library_path, pyside2_lib_test, "pyside2 library", find_dir=False)
+
+def find_pyside2_includes():
+    return find_path(include_path, pyside2_include_test, "pyside2 includes")
+
+def find_shiboken2_library():
+    return find_path(library_path, shiboken2_lib_test, "shiboken2 library", find_dir=False)
+
+def find_shiboken2_includes():
+    return find_path(include_path, shiboken2_include_test, "shiboken2 includes")
diff --git a/src/Gui/CMakeLists.txt b/src/Gui/CMakeLists.txt
index b709ecbcc..1348b7331 100644
--- a/src/Gui/CMakeLists.txt
+++ b/src/Gui/CMakeLists.txt
@@ -171,38 +171,27 @@ IF(OCULUS_FOUND)
     )
 ENDIF(OCULUS_FOUND)
 
-if(SHIBOKEN_INCLUDE_DIR)
+if(SHIBOKEN_FOUND)
     if (BUILD_QT5)
         add_definitions(-DHAVE_SHIBOKEN2)
     else()
         add_definitions(-DHAVE_SHIBOKEN)
     endif()
-    include_directories(
-        ${SHIBOKEN_INCLUDE_DIR}
-    )
     list(APPEND FreeCADGui_LIBS
         ${SHIBOKEN_LIBRARY}
     )
-endif(SHIBOKEN_INCLUDE_DIR)
+endif(SHIBOKEN_FOUND)
 
-if(PYSIDE_INCLUDE_DIR)
-    include_directories(
-        ${PYSIDE_INCLUDE_DIR}
-        ${PYSIDE_INCLUDE_DIR}/QtCore
-        ${PYSIDE_INCLUDE_DIR}/QtGui
-    )
+if(PYSIDE_FOUND)
     list(APPEND FreeCADGui_LIBS
         ${PYSIDE_LIBRARY}
     )
     if (BUILD_QT5)
-        include_directories(
-            ${PYSIDE_INCLUDE_DIR}/QtWidgets
-        )
         add_definitions(-DHAVE_PYSIDE2)
     else()
         add_definitions(-DHAVE_PYSIDE)
     endif()
-endif(PYSIDE_INCLUDE_DIR)
+endif(PYSIDE_FOUND)
 
 generate_from_xml(DocumentPy)
 generate_from_xml(PythonWorkbenchPy)
diff --git a/src/Mod/TechDraw/App/DrawViewPartPy.xml b/src/Mod/TechDraw/App/DrawViewPartPy.xml
index 91c7efe9d..2eb898e9d 100644
--- a/src/Mod/TechDraw/App/DrawViewPartPy.xml
+++ b/src/Mod/TechDraw/App/DrawViewPartPy.xml
@@ -102,16 +102,6 @@
       <Documentation>
         <UserDocu>formatGeometricEdge(index, style, weight, color, visible). Returns None.</UserDocu>
       </Documentation>
-    </Methode>
-        <Methode Name="getEdgeByIndex">
-      <Documentation>
-        <UserDocu>getEdgeByIndex(edgeIndex). Returns Part.TopoShape.</UserDocu>
-      </Documentation>
-    </Methode>
-    <Methode Name="getVertexByIndex">
-      <Documentation>
-        <UserDocu>getVertexByIndex(vertexIndex). Returns Part.TopoShape.</UserDocu>
-      </Documentation>
     </Methode>
     <CustomAttributes />
   </PythonExport>
diff --git a/src/Mod/TechDraw/App/DrawViewPartPyImp.cpp b/src/Mod/TechDraw/App/DrawViewPartPyImp.cpp
index c42d83673..1d543c4de 100644
--- a/src/Mod/TechDraw/App/DrawViewPartPyImp.cpp
+++ b/src/Mod/TechDraw/App/DrawViewPartPyImp.cpp
@@ -42,10 +42,6 @@
 #include <Base/Vector3D.h>
 #include <Base/VectorPy.h>
 
-#include <Mod/Part/App/TopoShape.h>
-#include <Mod/Part/App/TopoShapeEdgePy.h>
-#include <Mod/Part/App/TopoShapeVertexPy.h>
-
 #include "DrawViewPart.h"
 #include "GeometryObject.h"
 #include "Cosmetic.h"
@@ -68,7 +64,6 @@ std::string DrawViewPartPy::representation(void) const
 {
     return std::string("<DrawViewPart object>");
 }
-//TODO: gets & sets for geometry
 
 PyObject* DrawViewPartPy::clearCosmeticVertices(PyObject *args)
 {
@@ -508,33 +503,7 @@ PyObject* DrawViewPartPy::formatGeometricEdge(PyObject *args)
     return Py_None;
 }
 
-//------------------------------------------------------------------------------
-PyObject* DrawViewPartPy::getEdgeByIndex(PyObject *args)
-{
-    int edgeIndex = 0;
-    if (!PyArg_ParseTuple(args, "i", &edgeIndex)) {
-        throw Py::TypeError("expected (edgeIndex)");
-    }
-    DrawViewPart* dvp = getDrawViewPartPtr();
-    TechDraw::BaseGeom* geom = dvp->getGeomByIndex(edgeIndex);
-    TopoDS_Edge outEdge = geom->occEdge;
-    return new Part::TopoShapeEdgePy(new Part::TopoShape(outEdge));
-}
-
-PyObject* DrawViewPartPy::getVertexByIndex(PyObject *args)
-{
-    int vertexIndex = 0;
-    if (!PyArg_ParseTuple(args, "i", &vertexIndex)) {
-        throw Py::TypeError("expected (vertIndex)");
-    }
-    DrawViewPart* dvp = getDrawViewPartPtr();
-    TechDraw::Vertex* vert = dvp->getProjVertexByIndex(vertexIndex);
-    TopoDS_Vertex outVertex = vert->occVertex;
-    return new Part::TopoShapeVertexPy(new Part::TopoShape(outVertex));
-}
-
 
-//==============================================================================
 PyObject *DrawViewPartPy::getCustomAttributes(const char* /*attr*/) const
 {
     return 0;
diff --git a/src/Mod/TechDraw/Gui/Command.cpp b/src/Mod/TechDraw/Gui/Command.cpp
index b1c139591..8e8457865 100644
--- a/src/Mod/TechDraw/Gui/Command.cpp
+++ b/src/Mod/TechDraw/Gui/Command.cpp
@@ -67,7 +67,6 @@
 #include <Mod/TechDraw/App/DrawViewDraft.h>
 #include <Mod/TechDraw/App/DrawViewMulti.h>
 #include <Mod/TechDraw/App/DrawViewDetail.h>
-#include <Mod/TechDraw/App/DrawViewArch.h>
 #include <Mod/TechDraw/App/DrawUtil.h>
 #include <Mod/TechDraw/Gui/QGVPage.h>
 
@@ -1227,21 +1226,6 @@ void CmdTechDrawExportPageDxf::activated(int iMsg)
         return;
     }
 
-    std::vector<App::DocumentObject*> views = page->Views.getValues();
-    for (auto& v: views) {
-        if (v->isDerivedFrom(TechDraw::DrawViewArch::getClassTypeId())) {
-            QMessageBox::StandardButton rc =
-                QMessageBox::question(Gui::getMainWindow(), QObject::tr("Can not export selection"),
-                            QObject::tr("Page contains DrawViewArch which will not be exported. Continue?"),
-                            QMessageBox::StandardButtons(QMessageBox::Yes| QMessageBox::No));
-            if (rc == QMessageBox::No) {
-                return;
-            } else {
-                break;
-            }
-        }
-    }
-
 //WF? allow more than one TD Page per Dxf file??  1 TD page = 1 DXF file = 1 drawing?
     QString defaultDir;
     QString fileName = Gui::FileDialog::getSaveFileName(Gui::getMainWindow(),
diff --git a/src/Mod/TechDraw/Gui/DlgPrefsTechDraw.ui b/src/Mod/TechDraw/Gui/DlgPrefsTechDraw.ui
index 47a2a58e1..7cdec64c1 100644
--- a/src/Mod/TechDraw/Gui/DlgPrefsTechDraw.ui
+++ b/src/Mod/TechDraw/Gui/DlgPrefsTechDraw.ui
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>558</width>
-    <height>1110</height>
+    <height>1095</height>
    </rect>
   </property>
   <property name="windowTitle">
@@ -17,175 +17,225 @@
    <string>Default value for new Page</string>
   </property>
   <layout class="QGridLayout" name="gridLayout_5">
-   <item row="3" column="0">
-    <widget class="QGroupBox" name="gb_Templates">
+   <item row="0" column="0">
+    <widget class="QGroupBox" name="gbMisc">
      <property name="title">
-      <string>Files</string>
+      <string>General</string>
      </property>
-     <layout class="QVBoxLayout" name="verticalLayout_3">
+     <layout class="QVBoxLayout" name="verticalLayout">
       <item>
-       <layout class="QGridLayout" name="gridLayout_3">
+       <layout class="QFormLayout" name="formLayout">
+        <property name="fieldGrowthPolicy">
+         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
+        </property>
+        <item row="0" column="0">
+         <widget class="QLabel" name="lbl_Angle">
+          <property name="text">
+           <string>Projection Angle</string>
+          </property>
+         </widget>
+        </item>
         <item row="0" column="1">
-         <widget class="Gui::PrefFileChooser" name="pfc_DefTemp">
+         <widget class="Gui::PrefComboBox" name="cb_Angle">
           <property name="prefEntry" stdset="0">
-           <cstring>TemplateFile</cstring>
+           <cstring>ProjectionAngle</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/Files</cstring>
-          </property>
-         </widget>
-        </item>
-        <item row="4" column="0">
-         <widget class="QLabel" name="label_11">
-          <property name="text">
-           <string>Welding Directory</string>
+           <cstring>/Mod/TechDraw/General</cstring>
           </property>
+          <item>
+           <property name="text">
+            <string>First</string>
+           </property>
+          </item>
+          <item>
+           <property name="text">
+            <string>Third</string>
+           </property>
+          </item>
          </widget>
         </item>
         <item row="1" column="0">
-         <widget class="QLabel" name="label_2">
-          <property name="text">
-           <string>Template Directory</string>
-          </property>
-         </widget>
-        </item>
-        <item row="0" column="0">
-         <widget class="QLabel" name="label">
+         <widget class="QLabel" name="lbl_HidLine">
           <property name="text">
-           <string>Default Template</string>
+           <string>Hidden Line</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
-         <widget class="Gui::PrefFileChooser" name="pfc_DefDir">
-          <property name="mode">
-           <enum>Gui::FileChooser::Directory</enum>
+         <widget class="Gui::PrefComboBox" name="cb_HidLine">
+          <property name="toolTip">
+           <string>Style for hidden lines</string>
+          </property>
+          <property name="currentIndex">
+           <number>1</number>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>TemplateDir</cstring>
+           <cstring>HiddenLine</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/Files</cstring>
+           <cstring>Mod/TechDraw/General</cstring>
           </property>
+          <item>
+           <property name="text">
+            <string>NeverShow</string>
+           </property>
+          </item>
+          <item>
+           <property name="text">
+            <string>Solid</string>
+           </property>
+          </item>
+          <item>
+           <property name="text">
+            <string>Dash</string>
+           </property>
+          </item>
+          <item>
+           <property name="text">
+            <string>Dot</string>
+           </property>
+          </item>
+          <item>
+           <property name="text">
+            <string>DashDot</string>
+           </property>
+          </item>
+          <item>
+           <property name="text">
+            <string>DashDotDot</string>
+           </property>
+          </item>
          </widget>
         </item>
         <item row="2" column="0">
-         <widget class="QLabel" name="lbl_Hatch">
+         <widget class="Gui::PrefCheckBox" name="cb_Faces">
+          <property name="toolTip">
+           <string>Perform/skip face processing</string>
+          </property>
           <property name="text">
-           <string>Hatch Image</string>
+           <string>Detect Faces</string>
+          </property>
+          <property name="checked">
+           <bool>true</bool>
+          </property>
+          <property name="prefEntry" stdset="0">
+           <cstring>HandleFaces</cstring>
+          </property>
+          <property name="prefPath" stdset="0">
+           <cstring>/Mod/TechDraw/General</cstring>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
-         <widget class="QLabel" name="label_10">
+         <widget class="Gui::PrefCheckBox" name="cb_SectionEdges">
+          <property name="toolTip">
+           <string>Debugging option</string>
+          </property>
           <property name="text">
-           <string>Line Group File</string>
+           <string>Show Section Edges</string>
+          </property>
+          <property name="prefEntry" stdset="0">
+           <cstring>ShowSectionEdges</cstring>
+          </property>
+          <property name="prefPath" stdset="0">
+           <cstring>/Mod/TechDraw/General</cstring>
           </property>
          </widget>
         </item>
-        <item row="2" column="1">
-         <widget class="Gui::PrefFileChooser" name="pfc_HatchFile">
+        <item row="4" column="0">
+         <widget class="Gui::PrefCheckBox" name="cb_AutoDist">
           <property name="toolTip">
-           <string>Location of default svg/png fill file</string>
+           <string>Automatically distribute secondary views.</string>
+          </property>
+          <property name="text">
+           <string>AutoDistribute Secondary Views</string>
+          </property>
+          <property name="checked">
+           <bool>true</bool>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>FileHatch</cstring>
+           <cstring>AutoDist</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/Files</cstring>
+           <cstring>/Mod/TechDraw/General</cstring>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
-         <widget class="QLabel" name="label_9">
+         <widget class="QLabel" name="label_12">
+          <property name="font">
+           <font>
+            <weight>75</weight>
+            <bold>true</bold>
+           </font>
+          </property>
           <property name="text">
-           <string>PAT File</string>
+           <string>Drawing Updates</string>
           </property>
          </widget>
         </item>
-        <item row="5" column="1">
-         <widget class="Gui::PrefFileChooser" name="pfc_FilePattern">
+        <item row="6" column="0">
+         <widget class="Gui::PrefCheckBox" name="cb_Global">
           <property name="toolTip">
-           <string>Default location for PAT file</string>
-          </property>
-          <property name="prefEntry" stdset="0">
-           <cstring>FilePattern</cstring>
+           <string>Update drawings with changes in 3D model</string>
           </property>
-          <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/PAT</cstring>
+          <property name="text">
+           <string>Update with 3D (Global Policy)</string>
           </property>
-         </widget>
-        </item>
-        <item row="3" column="1">
-         <widget class="Gui::PrefFileChooser" name="pfc_LineGroup">
-          <property name="toolTip">
-           <string>Alternate Line Group file</string>
+          <property name="checked">
+           <bool>true</bool>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>LineGroupFile</cstring>
+           <cstring>GlobalUpdateDrawings</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/Files</cstring>
+           <cstring>Mod/TechDraw/General</cstring>
           </property>
          </widget>
         </item>
-        <item row="4" column="1">
-         <widget class="Gui::PrefFileChooser" name="pfc_Welding">
+        <item row="7" column="0">
+         <widget class="Gui::PrefCheckBox" name="cb_Override">
           <property name="toolTip">
-           <string>Default directory for welding symbols</string>
+           <string>Allow Page to override Global update setting</string>
           </property>
-          <property name="mode">
-           <enum>Gui::FileChooser::Directory</enum>
+          <property name="text">
+           <string>Allow Page Override</string>
+          </property>
+          <property name="checked">
+           <bool>true</bool>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>WeldingDir</cstring>
+           <cstring>AllowPageOverride</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/Files</cstring>
+           <cstring>Mod/TechDraw/General</cstring>
           </property>
          </widget>
         </item>
-       </layout>
-      </item>
-      <item>
-       <layout class="QGridLayout" name="gridLayout_4">
-        <item row="0" column="2">
-         <widget class="Gui::PrefLineEdit" name="le_NamePattern">
+        <item row="8" column="0">
+         <widget class="Gui::PrefCheckBox" name="cb_PageUpdate">
           <property name="toolTip">
-           <string>Default PAT pattern</string>
+           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Update Pages as scheduled or skip updates. This is the default setting for new Pages.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+          </property>
+          <property name="text">
+           <string>Keep Page Up to Date (default)</string>
+          </property>
+          <property name="checked">
+           <bool>true</bool>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>NamePattern</cstring>
+           <cstring>KeepPagesUpToDate</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/PAT</cstring>
-          </property>
-         </widget>
-        </item>
-        <item row="0" column="0">
-         <widget class="QLabel" name="label_8">
-          <property name="text">
-           <string>Pattern Name</string>
+           <cstring>Mod/TechDraw/General</cstring>
           </property>
          </widget>
         </item>
-        <item row="0" column="1">
-         <spacer name="horizontalSpacer_5">
-          <property name="orientation">
-           <enum>Qt::Horizontal</enum>
-          </property>
-          <property name="sizeHint" stdset="0">
-           <size>
-            <width>40</width>
-            <height>20</height>
-           </size>
-          </property>
-         </spacer>
-        </item>
        </layout>
       </item>
       <item>
-       <spacer name="verticalSpacer">
+       <spacer name="verticalSpacer_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
@@ -200,170 +250,12 @@
      </layout>
     </widget>
    </item>
-   <item row="2" column="0">
-    <widget class="QGroupBox" name="gb_Font">
-     <property name="sizePolicy">
-      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
-       <horstretch>0</horstretch>
-       <verstretch>0</verstretch>
-      </sizepolicy>
-     </property>
+   <item row="1" column="0">
+    <widget class="QGroupBox" name="gb_Colors">
      <property name="title">
-      <string>Labels</string>
+      <string>Colors</string>
      </property>
-     <layout class="QVBoxLayout" name="verticalLayout_4">
-      <item>
-       <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,0,0" columnstretch="0,0,0">
-        <item row="0" column="1">
-         <spacer name="horizontalSpacer_2">
-          <property name="orientation">
-           <enum>Qt::Horizontal</enum>
-          </property>
-          <property name="sizeType">
-           <enum>QSizePolicy::Preferred</enum>
-          </property>
-          <property name="sizeHint" stdset="0">
-           <size>
-            <width>40</width>
-            <height>20</height>
-           </size>
-          </property>
-         </spacer>
-        </item>
-        <item row="0" column="0">
-         <widget class="QLabel" name="lbl_LabelFont">
-          <property name="text">
-           <string>Label Font</string>
-          </property>
-         </widget>
-        </item>
-        <item row="1" column="0">
-         <widget class="QLabel" name="label_6">
-          <property name="text">
-           <string>Label Size</string>
-          </property>
-         </widget>
-        </item>
-        <item row="0" column="2">
-         <widget class="Gui::PrefFontBox" name="pfb_LabelFont">
-          <property name="toolTip">
-           <string>Font for View Labels</string>
-          </property>
-          <property name="currentFont">
-           <font>
-            <family>DejaVu Sans</family>
-           </font>
-          </property>
-          <property name="prefEntry" stdset="0">
-           <cstring>LabelFont</cstring>
-          </property>
-          <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/Labels</cstring>
-          </property>
-         </widget>
-        </item>
-        <item row="2" column="0">
-         <widget class="QLabel" name="lbl_TemplateDot">
-          <property name="text">
-           <string>Editable Text Marker Size</string>
-          </property>
-         </widget>
-        </item>
-        <item row="1" column="1">
-         <spacer name="horizontalSpacer_6">
-          <property name="orientation">
-           <enum>Qt::Horizontal</enum>
-          </property>
-          <property name="sizeHint" stdset="0">
-           <size>
-            <width>40</width>
-            <height>20</height>
-           </size>
-          </property>
-         </spacer>
-        </item>
-        <item row="2" column="1">
-         <spacer name="horizontalSpacer_7">
-          <property name="orientation">
-           <enum>Qt::Horizontal</enum>
-          </property>
-          <property name="sizeHint" stdset="0">
-           <size>
-            <width>40</width>
-            <height>20</height>
-           </size>
-          </property>
-         </spacer>
-        </item>
-        <item row="1" column="2">
-         <widget class="Gui::PrefUnitSpinBox" name="plsb_LabelSize">
-          <property name="toolTip">
-           <string>View Label size in units</string>
-          </property>
-          <property name="value">
-           <double>8.000000000000000</double>
-          </property>
-          <property name="prefEntry" stdset="0">
-           <cstring>LabelSize</cstring>
-          </property>
-          <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/Labels</cstring>
-          </property>
-         </widget>
-        </item>
-        <item row="2" column="2">
-         <widget class="Gui::PrefUnitSpinBox" name="plsb_TemplateDot">
-          <property name="toolTip">
-           <string>Size of editable text marker in Templates (green dot).</string>
-          </property>
-          <property name="value">
-           <double>3.000000000000000</double>
-          </property>
-          <property name="prefEntry" stdset="0">
-           <cstring>TemplateDotSize</cstring>
-          </property>
-          <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/General</cstring>
-          </property>
-         </widget>
-        </item>
-       </layout>
-      </item>
-      <item>
-       <spacer name="verticalSpacer_5">
-        <property name="orientation">
-         <enum>Qt::Vertical</enum>
-        </property>
-        <property name="sizeHint" stdset="0">
-         <size>
-          <width>20</width>
-          <height>40</height>
-         </size>
-        </property>
-       </spacer>
-      </item>
-     </layout>
-    </widget>
-   </item>
-   <item row="4" column="0">
-    <spacer name="verticalSpacer_2">
-     <property name="orientation">
-      <enum>Qt::Vertical</enum>
-     </property>
-     <property name="sizeHint" stdset="0">
-      <size>
-       <width>20</width>
-       <height>40</height>
-      </size>
-     </property>
-    </spacer>
-   </item>
-   <item row="1" column="0">
-    <widget class="QGroupBox" name="gb_Colors">
-     <property name="title">
-      <string>Colors</string>
-     </property>
-     <layout class="QVBoxLayout" name="verticalLayout_2">
+     <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="2" column="4">
@@ -382,9 +274,6 @@
         </item>
         <item row="3" column="2">
          <widget class="Gui::PrefColorButton" name="pcb_Background">
-          <property name="toolTip">
-           <string>Window background color</string>
-          </property>
           <property name="color">
            <color>
             <red>80</red>
@@ -402,9 +291,6 @@
         </item>
         <item row="1" column="2">
          <widget class="Gui::PrefColorButton" name="pcb_PreSelect">
-          <property name="toolTip">
-           <string>Preselection color</string>
-          </property>
           <property name="color">
            <color>
             <red>255</red>
@@ -429,9 +315,6 @@
         </item>
         <item row="0" column="6">
          <widget class="Gui::PrefColorButton" name="pcb_Hidden">
-          <property name="toolTip">
-           <string>Hidden line color</string>
-          </property>
           <property name="prefEntry" stdset="0">
            <cstring>HiddenColor</cstring>
           </property>
@@ -455,9 +338,6 @@
         </item>
         <item row="2" column="2">
          <widget class="Gui::PrefColorButton" name="pcb_Select">
-          <property name="toolTip">
-           <string>Selected item color</string>
-          </property>
           <property name="color">
            <color>
             <red>28</red>
@@ -488,9 +368,6 @@
         </item>
         <item row="0" column="2">
          <widget class="Gui::PrefColorButton" name="pcb_Normal">
-          <property name="toolTip">
-           <string>Normal lne color</string>
-          </property>
           <property name="color">
            <color>
             <red>0</red>
@@ -535,9 +412,6 @@
         </item>
         <item row="1" column="6">
          <widget class="Gui::PrefColorButton" name="pcb_Surface">
-          <property name="toolTip">
-           <string>Section Face color</string>
-          </property>
           <property name="color">
            <color>
             <red>225</red>
@@ -562,9 +436,6 @@
         </item>
         <item row="2" column="6">
          <widget class="Gui::PrefColorButton" name="pcb_Hatch">
-          <property name="toolTip">
-           <string>Section Face Hatch color</string>
-          </property>
           <property name="color">
            <color>
             <red>0</red>
@@ -595,10 +466,7 @@
          </widget>
         </item>
         <item row="3" column="6">
-         <widget class="Gui::PrefColorButton" name="pcb_GeomHatch">
-          <property name="toolTip">
-           <string>Default geometric hatch color</string>
-          </property>
+         <widget class="Gui::PrefColorButton" name="colorButton">
           <property name="color">
            <color>
             <red>0</red>
@@ -614,42 +482,6 @@
           </property>
          </widget>
         </item>
-        <item row="4" column="2">
-         <widget class="Gui::PrefColorButton" name="pcb_Face">
-          <property name="toolTip">
-           <string>Default Face Color</string>
-          </property>
-          <property name="color">
-           <color>
-            <red>255</red>
-            <green>255</green>
-            <blue>255</blue>
-           </color>
-          </property>
-          <property name="prefEntry" stdset="0">
-           <cstring>FaceColor</cstring>
-          </property>
-          <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/Colors</cstring>
-          </property>
-         </widget>
-        </item>
-        <item row="4" column="0">
-         <widget class="Gui::PrefCheckBox" name="pcb_PaintFaces">
-          <property name="toolTip">
-           <string>Transparent faces if checked</string>
-          </property>
-          <property name="text">
-           <string>Transparent Faces</string>
-          </property>
-          <property name="prefEntry" stdset="0">
-           <cstring>ClearFace</cstring>
-          </property>
-          <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/Colors</cstring>
-          </property>
-         </widget>
-        </item>
        </layout>
       </item>
       <item>
@@ -668,225 +500,320 @@
      </layout>
     </widget>
    </item>
-   <item row="0" column="0">
-    <widget class="QGroupBox" name="gbMisc">
+   <item row="3" column="0">
+    <widget class="QGroupBox" name="gb_Templates">
      <property name="title">
-      <string>General</string>
+      <string>Files</string>
      </property>
-     <layout class="QVBoxLayout" name="verticalLayout">
+     <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
-       <layout class="QFormLayout" name="formLayout">
-        <property name="fieldGrowthPolicy">
-         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
-        </property>
-        <item row="0" column="0">
-         <widget class="QLabel" name="lbl_Angle">
-          <property name="text">
-           <string>Projection Angle</string>
-          </property>
-         </widget>
-        </item>
+       <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="1">
-         <widget class="Gui::PrefComboBox" name="cb_Angle">
+         <widget class="Gui::PrefFileChooser" name="pfc_DefTemp">
           <property name="prefEntry" stdset="0">
-           <cstring>ProjectionAngle</cstring>
+           <cstring>TemplateFile</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/General</cstring>
+           <cstring>/Mod/TechDraw/Files</cstring>
+          </property>
+         </widget>
+        </item>
+        <item row="4" column="0">
+         <widget class="QLabel" name="label_11">
+          <property name="text">
+           <string>Welding Directory</string>
           </property>
-          <item>
-           <property name="text">
-            <string>First</string>
-           </property>
-          </item>
-          <item>
-           <property name="text">
-            <string>Third</string>
-           </property>
-          </item>
          </widget>
         </item>
         <item row="1" column="0">
-         <widget class="QLabel" name="lbl_HidLine">
+         <widget class="QLabel" name="label_2">
           <property name="text">
-           <string>Hidden Line</string>
+           <string>Template Directory</string>
           </property>
          </widget>
         </item>
-        <item row="1" column="1">
-         <widget class="Gui::PrefComboBox" name="cb_HidLine">
-          <property name="toolTip">
-           <string>Style for hidden lines</string>
+        <item row="0" column="0">
+         <widget class="QLabel" name="label">
+          <property name="text">
+           <string>Default Template</string>
           </property>
-          <property name="currentIndex">
-           <number>1</number>
+         </widget>
+        </item>
+        <item row="1" column="1">
+         <widget class="Gui::PrefFileChooser" name="pfc_DefDir">
+          <property name="mode">
+           <enum>Gui::FileChooser::Directory</enum>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>HiddenLine</cstring>
+           <cstring>TemplateDir</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>Mod/TechDraw/General</cstring>
+           <cstring>/Mod/TechDraw/Files</cstring>
           </property>
-          <item>
-           <property name="text">
-            <string>NeverShow</string>
-           </property>
-          </item>
-          <item>
-           <property name="text">
-            <string>Solid</string>
-           </property>
-          </item>
-          <item>
-           <property name="text">
-            <string>Dash</string>
-           </property>
-          </item>
-          <item>
-           <property name="text">
-            <string>Dot</string>
-           </property>
-          </item>
-          <item>
-           <property name="text">
-            <string>DashDot</string>
-           </property>
-          </item>
-          <item>
-           <property name="text">
-            <string>DashDotDot</string>
-           </property>
-          </item>
          </widget>
         </item>
         <item row="2" column="0">
-         <widget class="Gui::PrefCheckBox" name="cb_Faces">
-          <property name="toolTip">
-           <string>Perform/skip face processing</string>
+         <widget class="QLabel" name="lbl_Hatch">
+          <property name="text">
+           <string>Hatch Image</string>
           </property>
+         </widget>
+        </item>
+        <item row="3" column="0">
+         <widget class="QLabel" name="label_10">
           <property name="text">
-           <string>Detect Faces</string>
+           <string>Line Group File</string>
           </property>
-          <property name="checked">
-           <bool>true</bool>
+         </widget>
+        </item>
+        <item row="2" column="1">
+         <widget class="Gui::PrefFileChooser" name="pfc_HatchFile">
+          <property name="toolTip">
+           <string>Location of default svg/png fill file</string>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>HandleFaces</cstring>
+           <cstring>FileHatch</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/General</cstring>
+           <cstring>/Mod/TechDraw/Files</cstring>
           </property>
          </widget>
         </item>
-        <item row="3" column="0">
-         <widget class="Gui::PrefCheckBox" name="cb_SectionEdges">
-          <property name="toolTip">
-           <string>Debugging option</string>
-          </property>
+        <item row="5" column="0">
+         <widget class="QLabel" name="label_9">
           <property name="text">
-           <string>Show Section Edges</string>
+           <string>PAT File</string>
+          </property>
+         </widget>
+        </item>
+        <item row="5" column="1">
+         <widget class="Gui::PrefFileChooser" name="pfc_FilePattern">
+          <property name="toolTip">
+           <string>Default location for PAT file</string>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>ShowSectionEdges</cstring>
+           <cstring>FilePattern</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/General</cstring>
+           <cstring>/Mod/TechDraw/PAT</cstring>
           </property>
          </widget>
         </item>
-        <item row="4" column="0">
-         <widget class="Gui::PrefCheckBox" name="cb_AutoDist">
+        <item row="3" column="1">
+         <widget class="Gui::PrefFileChooser" name="pfc_LineGroup">
           <property name="toolTip">
-           <string>Automatically distribute secondary views.</string>
+           <string>Alternate Line Group file</string>
           </property>
-          <property name="text">
-           <string>AutoDistribute Secondary Views</string>
+          <property name="prefEntry" stdset="0">
+           <cstring>LineGroupFile</cstring>
           </property>
-          <property name="checked">
-           <bool>true</bool>
+          <property name="prefPath" stdset="0">
+           <cstring>/Mod/TechDraw/Files</cstring>
+          </property>
+         </widget>
+        </item>
+        <item row="4" column="1">
+         <widget class="Gui::PrefFileChooser" name="pfc_Welding">
+          <property name="toolTip">
+           <string>Default directory for welding symbols</string>
+          </property>
+          <property name="mode">
+           <enum>Gui::FileChooser::Directory</enum>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>AutoDist</cstring>
+           <cstring>WeldingDir</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>/Mod/TechDraw/General</cstring>
+           <cstring>/Mod/TechDraw/Files</cstring>
           </property>
          </widget>
         </item>
-        <item row="5" column="0">
-         <widget class="QLabel" name="label_12">
-          <property name="font">
-           <font>
-            <weight>75</weight>
-            <bold>true</bold>
-           </font>
+       </layout>
+      </item>
+      <item>
+       <layout class="QGridLayout" name="gridLayout_4">
+        <item row="0" column="2">
+         <widget class="Gui::PrefLineEdit" name="le_NamePattern">
+          <property name="toolTip">
+           <string>Default PAT pattern</string>
+          </property>
+          <property name="prefEntry" stdset="0">
+           <cstring>NamePattern</cstring>
           </property>
+          <property name="prefPath" stdset="0">
+           <cstring>/Mod/TechDraw/PAT</cstring>
+          </property>
+         </widget>
+        </item>
+        <item row="0" column="0">
+         <widget class="QLabel" name="label_8">
           <property name="text">
-           <string>Drawing Updates</string>
+           <string>Pattern Name</string>
           </property>
          </widget>
         </item>
-        <item row="6" column="0">
-         <widget class="Gui::PrefCheckBox" name="cb_Global">
-          <property name="toolTip">
-           <string>Update drawings with changes in 3D model</string>
+        <item row="0" column="1">
+         <spacer name="horizontalSpacer_5">
+          <property name="orientation">
+           <enum>Qt::Horizontal</enum>
+          </property>
+          <property name="sizeHint" stdset="0">
+           <size>
+            <width>40</width>
+            <height>20</height>
+           </size>
           </property>
+         </spacer>
+        </item>
+       </layout>
+      </item>
+      <item>
+       <spacer name="verticalSpacer">
+        <property name="orientation">
+         <enum>Qt::Vertical</enum>
+        </property>
+        <property name="sizeHint" stdset="0">
+         <size>
+          <width>20</width>
+          <height>40</height>
+         </size>
+        </property>
+       </spacer>
+      </item>
+     </layout>
+    </widget>
+   </item>
+   <item row="2" column="0">
+    <widget class="QGroupBox" name="gb_Font">
+     <property name="sizePolicy">
+      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
+       <horstretch>0</horstretch>
+       <verstretch>0</verstretch>
+      </sizepolicy>
+     </property>
+     <property name="title">
+      <string>Labels</string>
+     </property>
+     <layout class="QVBoxLayout" name="verticalLayout_4">
+      <item>
+       <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,0,0" columnstretch="0,0,0">
+        <item row="0" column="1">
+         <spacer name="horizontalSpacer_2">
+          <property name="orientation">
+           <enum>Qt::Horizontal</enum>
+          </property>
+          <property name="sizeType">
+           <enum>QSizePolicy::Preferred</enum>
+          </property>
+          <property name="sizeHint" stdset="0">
+           <size>
+            <width>40</width>
+            <height>20</height>
+           </size>
+          </property>
+         </spacer>
+        </item>
+        <item row="0" column="0">
+         <widget class="QLabel" name="lbl_LabelFont">
           <property name="text">
-           <string>Update with 3D (Global Policy)</string>
+           <string>Label Font</string>
           </property>
-          <property name="checked">
-           <bool>true</bool>
+         </widget>
+        </item>
+        <item row="1" column="0">
+         <widget class="QLabel" name="label_6">
+          <property name="text">
+           <string>Label Size</string>
+          </property>
+         </widget>
+        </item>
+        <item row="0" column="2">
+         <widget class="Gui::PrefFontBox" name="pfb_LabelFont">
+          <property name="toolTip">
+           <string>Font for View Labels</string>
+          </property>
+          <property name="currentFont">
+           <font>
+            <family>DejaVu Sans</family>
+           </font>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>GlobalUpdateDrawings</cstring>
+           <cstring>LabelFont</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>Mod/TechDraw/General</cstring>
+           <cstring>/Mod/TechDraw/Labels</cstring>
           </property>
          </widget>
         </item>
-        <item row="7" column="0">
-         <widget class="Gui::PrefCheckBox" name="cb_Override">
-          <property name="toolTip">
-           <string>Allow Page to override Global update setting</string>
-          </property>
+        <item row="2" column="0">
+         <widget class="QLabel" name="lbl_TemplateDot">
           <property name="text">
-           <string>Allow Page Override</string>
+           <string>Editable Text Marker Size</string>
           </property>
-          <property name="checked">
-           <bool>true</bool>
+         </widget>
+        </item>
+        <item row="1" column="1">
+         <spacer name="horizontalSpacer_6">
+          <property name="orientation">
+           <enum>Qt::Horizontal</enum>
+          </property>
+          <property name="sizeHint" stdset="0">
+           <size>
+            <width>40</width>
+            <height>20</height>
+           </size>
+          </property>
+         </spacer>
+        </item>
+        <item row="2" column="1">
+         <spacer name="horizontalSpacer_7">
+          <property name="orientation">
+           <enum>Qt::Horizontal</enum>
+          </property>
+          <property name="sizeHint" stdset="0">
+           <size>
+            <width>40</width>
+            <height>20</height>
+           </size>
+          </property>
+         </spacer>
+        </item>
+        <item row="1" column="2">
+         <widget class="Gui::PrefUnitSpinBox" name="plsb_LabelSize">
+          <property name="toolTip">
+           <string>View Label size in units</string>
+          </property>
+          <property name="value">
+           <double>8.000000000000000</double>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>AllowPageOverride</cstring>
+           <cstring>LabelSize</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>Mod/TechDraw/General</cstring>
+           <cstring>/Mod/TechDraw/Labels</cstring>
           </property>
          </widget>
         </item>
-        <item row="8" column="0">
-         <widget class="Gui::PrefCheckBox" name="cb_PageUpdate">
+        <item row="2" column="2">
+         <widget class="Gui::PrefUnitSpinBox" name="plsb_TemplateDot">
           <property name="toolTip">
-           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Update Pages as scheduled or skip updates. This is the default setting for new Pages.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
-          </property>
-          <property name="text">
-           <string>Keep Page Up to Date (default)</string>
+           <string>Size of editable text marker in Templates (green dot).</string>
           </property>
-          <property name="checked">
-           <bool>true</bool>
+          <property name="value">
+           <double>3.000000000000000</double>
           </property>
           <property name="prefEntry" stdset="0">
-           <cstring>KeepPagesUpToDate</cstring>
+           <cstring>TemplateDotSize</cstring>
           </property>
           <property name="prefPath" stdset="0">
-           <cstring>Mod/TechDraw/General</cstring>
+           <cstring>/Mod/TechDraw/General</cstring>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
-       <spacer name="verticalSpacer_4">
+       <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
@@ -901,6 +828,19 @@
      </layout>
     </widget>
    </item>
+   <item row="4" column="0">
+    <spacer name="verticalSpacer_2">
+     <property name="orientation">
+      <enum>Qt::Vertical</enum>
+     </property>
+     <property name="sizeHint" stdset="0">
+      <size>
+       <width>20</width>
+       <height>40</height>
+      </size>
+     </property>
+    </spacer>
+   </item>
   </layout>
  </widget>
  <customwidgets>
diff --git a/src/Mod/TechDraw/Gui/DlgPrefsTechDrawImp.cpp b/src/Mod/TechDraw/Gui/DlgPrefsTechDrawImp.cpp
index c614e1024..53be57bdc 100644
--- a/src/Mod/TechDraw/Gui/DlgPrefsTechDrawImp.cpp
+++ b/src/Mod/TechDraw/Gui/DlgPrefsTechDrawImp.cpp
@@ -61,8 +61,6 @@ void DlgPrefsTechDrawImp::saveSettings()
     pcb_Surface->onSave();
     pcb_Background->onSave();
     pcb_Hatch->onSave();
-    pcb_Face->onSave();
-    pcb_PaintFaces->onSave();  //check box!
 
     pfb_LabelFont->onSave();
     plsb_LabelSize->onSave();
@@ -96,8 +94,6 @@ void DlgPrefsTechDrawImp::loadSettings()
     pcb_Surface->onRestore();
     pcb_Background->onRestore();
     pcb_Hatch->onRestore();
-    pcb_Face->onRestore();
-    pcb_PaintFaces->onRestore();  //check box!
 
     pfb_LabelFont->onRestore();
     plsb_LabelSize->onRestore();
diff --git a/src/Mod/TechDraw/Gui/QGIFace.cpp b/src/Mod/TechDraw/Gui/QGIFace.cpp
index 92943b397..42fcb63a1 100644
--- a/src/Mod/TechDraw/Gui/QGIFace.cpp
+++ b/src/Mod/TechDraw/Gui/QGIFace.cpp
@@ -72,7 +72,7 @@ QGIFace::QGIFace(int index) :
     projIndex(index)
 {
     m_segCount = 0;
-//    setFillMode(NoFill);
+    setFillMode(NoFill);
     isHatched(false);
     setFlag(QGraphicsItem::ItemClipsChildrenToShape,true);
 
@@ -91,23 +91,17 @@ QGIFace::QGIFace(int index) :
 
     m_svgCol = SVGCOLDEFAULT;
     m_fillScale = 1.0;
-
-    m_colDefFill = Qt::white;
-    m_styleDef = Qt::SolidPattern;
-    m_styleSelect = Qt::SolidPattern;
-
+    
     getParameters();
  
     m_styleNormal = m_styleDef;
     m_fillStyle = m_styleDef;
     m_fill = m_styleDef;
-    if (m_defClearFace) {
-        setFillMode(NoFill);
-        setFill(Qt::transparent, m_styleDef);
-    } else {
-        setFillMode(PlainFill);
-        setFill(m_colNormalFill, m_styleDef);
-    }
+    m_colDefFill = Qt::white;
+    m_colNormalFill = m_colDefFill;
+    m_styleDef = Qt::SolidPattern;
+    m_styleSelect = Qt::SolidPattern;
+
 }
 
 QGIFace::~QGIFace()
@@ -157,10 +151,6 @@ void QGIFace::draw()
                     m_brush.setTexture(m_texture);
                 }
             }
-        } else if (m_mode == PlainFill) {
-            if (!m_lineSets.empty()) {
-                setFill(m_colNormalFill, m_styleNormal);
-            }
         }
     }
     show();
@@ -176,7 +166,8 @@ void QGIFace::setPrettyNormal() {
         m_fillStyle = m_styleNormal;
         m_fill = m_styleNormal;
         m_brush.setTexture(QPixmap());
-        m_brush.setStyle(m_fill);
+//        m_brush.setStyle(m_fillStyle);
+        m_brush.setStyle(m_fill);   //???
         m_fillColor = m_colNormalFill;
     }
     QGIPrimPath::setPrettyNormal();
@@ -629,17 +620,9 @@ void QGIFace::getParameters(void)
     hGrp = App::GetApplication().GetUserParameter()
         .GetGroup("BaseApp")->GetGroup("Preferences")->GetGroup("Mod/TechDraw/Decorations");
     m_maxTile = hGrp->GetInt("MaxSVGTile",10000l);
-
-    hGrp = App::GetApplication().GetUserParameter()
-        .GetGroup("BaseApp")->GetGroup("Preferences")->GetGroup("Mod/TechDraw/Colors");
-    App::Color temp = hGrp->GetUnsigned("FaceColor",0xffffffff);
-    setFillColor(temp.asValue<QColor>());
-
-    hGrp = App::GetApplication().GetUserParameter()
-        .GetGroup("BaseApp")->GetGroup("Preferences")->GetGroup("Mod/TechDraw/Colors");
-    m_defClearFace = hGrp->GetBool("ClearFace",false);
 }
 
+
 QRectF QGIFace::boundingRect() const
 {
     return shape().controlPointRect();
@@ -655,6 +638,9 @@ void QGIFace::paint ( QPainter * painter, const QStyleOptionGraphicsItem * optio
     myOption.state &= ~QStyle::State_Selected;
 //    painter->drawRect(boundingRect());          //good for debugging
 
+//    m_brush.setStyle(m_fillStyle);
+//    m_brush.setColor(m_fillColor);
+//    setBrush(m_brush);
     QGIPrimPath::paint (painter, &myOption, widget);
 }
 
diff --git a/src/Mod/TechDraw/Gui/QGIFace.h b/src/Mod/TechDraw/Gui/QGIFace.h
index 97b81cf7d..5e3635a89 100644
--- a/src/Mod/TechDraw/Gui/QGIFace.h
+++ b/src/Mod/TechDraw/Gui/QGIFace.h
@@ -85,6 +85,11 @@ public:
     bool isHatched(void) {return m_isHatched;}
     void setFillMode(fillMode m);
 
+/*    //plain color fill parms*/
+/*    void setFill(QColor c, Qt::BrushStyle s);*/
+/*    void setFill(QBrush b);*/
+/*    void resetFill();*/
+
     //general hatch parms & methods
     void setHatchColor(App::Color c);
     void setHatchScale(double s);
@@ -146,6 +151,16 @@ protected:
 
 
 private:
+/*    QBrush m_brush;*/
+/*    Qt::BrushStyle m_fillStyle;                 //current fill style*/
+/*    QColor m_fillColor;                         //current fill color*/
+
+/*    QColor m_colDefFill;                        //"no color" default normal fill color*/
+/*    QColor m_colNormalFill;                     //current Normal fill color*/
+/*    Qt::BrushStyle m_styleDef;                  //default Normal fill style*/
+/*    Qt::BrushStyle m_styleNormal;               //current Normal fill style*/
+/*    Qt::BrushStyle m_styleSelect;               //Select/preSelect fill style*/
+ 
     QPixmap m_texture;                          //
  
     QPainterPath m_outline;                     //
@@ -154,8 +169,6 @@ private:
  
     QColor m_geomColor;                        //color for crosshatch lines
     double m_geomWeight;                       //lineweight for crosshatch lines
-    bool m_defClearFace;
-    QColor m_defFaceColor;
 };
 
 }
diff --git a/src/Mod/TechDraw/Gui/QGIPrimPath.cpp b/src/Mod/TechDraw/Gui/QGIPrimPath.cpp
index 30a94c1f6..0fb4ab283 100644
--- a/src/Mod/TechDraw/Gui/QGIPrimPath.cpp
+++ b/src/Mod/TechDraw/Gui/QGIPrimPath.cpp
@@ -138,10 +138,9 @@ void QGIPrimPath::setPrettySel() {
     update();
 }
 
-//wf: why would a face use it's parent's normal colour?
-//this always goes to parameter
 QColor QGIPrimPath::getNormalColor()
 {
+
     QColor result;
     QGIView *parent;
 
@@ -165,7 +164,6 @@ QColor QGIPrimPath::getNormalColor()
         fcColor.setPackedValue(hGrp->GetUnsigned("NormalColor", 0x00000000));
         result = fcColor.asValue<QColor>();
     }
-
     return result;
 }
 
@@ -284,13 +282,13 @@ void QGIPrimPath::mousePressEvent(QGraphicsSceneMouseEvent * event)
 }
 
 void QGIPrimPath::setFill(QColor c, Qt::BrushStyle s) {
-    setFillColor(c);
+    m_colNormalFill = c;
     m_styleNormal = s;
     m_fill = s;
 }
 
 void QGIPrimPath::setFill(QBrush b) {
-    setFillColor(b.color());
+    m_colNormalFill = b.color();
     m_styleNormal = b.style();
     m_fill = b.style();
 }
@@ -301,13 +299,6 @@ void QGIPrimPath::resetFill() {
     m_fill = m_styleDef;
 }
 
-void QGIPrimPath::setFillColor(QColor c)
-{ 
-    m_colNormalFill = c;
-    m_colDefFill = c;
-}
-
-
 void QGIPrimPath::paint ( QPainter * painter, const QStyleOptionGraphicsItem * option, QWidget * widget) {
     QStyleOptionGraphicsItem myOption(*option);
     myOption.state &= ~QStyle::State_Selected;
@@ -317,7 +308,7 @@ void QGIPrimPath::paint ( QPainter * painter, const QStyleOptionGraphicsItem * o
     m_pen.setStyle(m_styleCurrent);
     setPen(m_pen);
 
-    m_brush.setColor(m_colNormalFill);
+    m_brush.setColor(m_fillColor);  //pencolr
     m_brush.setStyle(m_fill);
     setBrush(m_brush);
 
diff --git a/src/Mod/TechDraw/Gui/QGIPrimPath.h b/src/Mod/TechDraw/Gui/QGIPrimPath.h
index d23fe6fbb..0625bf557 100644
--- a/src/Mod/TechDraw/Gui/QGIPrimPath.h
+++ b/src/Mod/TechDraw/Gui/QGIPrimPath.h
@@ -58,18 +58,15 @@ public:
     void setStyle(int s);
     virtual void setNormalColor(QColor c);
     virtual void setCapStyle(Qt::PenCapStyle c);
+    Qt::BrushStyle getFill() { return m_fill; }
 
     //plain color fill parms
     void setFill(Qt::BrushStyle f) { m_fill = f; }
-    Qt::BrushStyle getFill() { return m_fill; }
-
     void setFill(QColor c, Qt::BrushStyle s);
     void setFill(QBrush b);
     void resetFill();
-/*    void setFillColor(QColor c) { m_colNormalFill = c;*/
-/*                                  m_colDefFill = c; }*/
-    void setFillColor(QColor c);
-    QColor getFillColor(void) { return m_colDefFill; }
+    void setFillColor(QColor c) { m_colNormalFill = c;
+                                  m_colDefFill = c; }
 
 protected:
     virtual void hoverEnterEvent(QGraphicsSceneHoverEvent *event) override;
diff --git a/src/Mod/TechDraw/Gui/QGIViewPart.cpp b/src/Mod/TechDraw/Gui/QGIViewPart.cpp
index 23c5b6b39..02c106735 100644
--- a/src/Mod/TechDraw/Gui/QGIViewPart.cpp
+++ b/src/Mod/TechDraw/Gui/QGIViewPart.cpp
@@ -372,7 +372,6 @@ void QGIViewPart::drawViewPart()
             QGIFace* newFace = drawFace(*fit,i);
             newFace->isHatched(false);
             newFace->setFillMode(QGIFace::PlainFill);
-//            newFace->setFill(QColor(Qt::red), Qt::SolidPattern);  //this overrides the QGIF defaults
             TechDraw::DrawHatch* fHatch = faceIsHatched(i,hatchObjs);
             TechDraw::DrawGeomHatch* fGeom = faceIsGeomHatched(i,geomObjs);
             if (fGeom) {
@@ -419,8 +418,8 @@ void QGIViewPart::drawViewPart()
             bool drawEdges = prefFaceEdges();
             newFace->setDrawEdges(drawEdges);                                        //pref. for debugging only
             newFace->setZValue(ZVALUE::FACE);
-            newFace->setPrettyNormal();
             newFace->draw();
+            newFace->setPrettyNormal();
         }
     }
 #endif //#if MOD_TECHDRAW_HANDLE_FACES
