diff --git a/cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake b/cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake
index ee0a6142f..a8cde118f 100644
--- a/cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake
+++ b/cMake/FreeCAD_Helpers/SetupShibokenAndPyside.cmake
@@ -43,6 +43,20 @@ macro(SetupShibokenAndPyside)
             endif()
         endif()
 
+        # pyside2 changed it's cmake files, this is the dance we have
+        # to dance to be compatible with the old (<5.12) and the new versions (>=5.12)
+        if(NOT SHIBOKEN_LIBRARY)
+            if(TARGET Shiboken2::libshiboken)
+                get_property(SHIBOKEN_LIBRARY TARGET Shiboken2::libshiboken PROPERTY IMPORTED_LOCATION_RELEASE)
+            endif(TARGET Shiboken2::libshiboken)
+        endif(NOT SHIBOKEN_LIBRARY)
+
+        if(NOT SHIBOKEN_LIBRARY)
+            message("====================\n"
+                    "shiboken2 not found.\n"
+                    "====================\n")
+        endif(NOT SHIBOKEN_LIBRARY)
+
         if(NOT SHIBOKEN_INCLUDE_DIR)
             message("====================\n"
                     "shiboken2 not found.\n"
