diff --git a/src/Mod/Fem/App/HypothesisPy.cpp b/src/Mod/Fem/App/HypothesisPy.cpp
index 7cebb2dcc..877bf0c08 100755
--- a/src/Mod/Fem/App/HypothesisPy.cpp
+++ b/src/Mod/Fem/App/HypothesisPy.cpp
@@ -21,6 +21,7 @@
  ***************************************************************************/
 
 #include "PreCompiled.h"
+#include "HypothesisPy.h"
 
 #ifndef _PreComp_
 # include <sstream>
@@ -62,7 +63,6 @@
 # endif
 #endif
 
-#include "HypothesisPy.h"
 #include "FemMeshPy.h"
 #include <Base/Interpreter.h>
 #include <Mod/Part/App/TopoShapePy.h>
