diff --git a/src/Mod/Sketcher/App/GeoEnum.cpp b/src/Mod/Sketcher/App/GeoEnum.cpp
index cb3cd4f83d..06b1c385b0 100644
--- a/src/Mod/Sketcher/App/GeoEnum.cpp
+++ b/src/Mod/Sketcher/App/GeoEnum.cpp
@@ -35,3 +35,8 @@ bool GeoElementId::operator!=(const GeoElementId& obj) const
 {
     return this->GeoId != obj.GeoId || this->Pos != obj.Pos;
 }
+
+
+inline constexpr const GeoElementId GeoElementId::RtPnt = GeoElementId(GeoEnum::RtPnt, PointPos::start);
+inline constexpr const GeoElementId GeoElementId::HAxis = GeoElementId(GeoEnum::HAxis, PointPos::none);
+inline constexpr const GeoElementId GeoElementId::VAxis = GeoElementId(GeoEnum::VAxis, PointPos::end);
\ No newline at end of file
diff --git a/src/Mod/Sketcher/App/GeoEnum.h b/src/Mod/Sketcher/App/GeoEnum.h
index df594c3a62..0173fa989e 100644
--- a/src/Mod/Sketcher/App/GeoEnum.h
+++ b/src/Mod/Sketcher/App/GeoEnum.h
@@ -96,7 +96,7 @@ enum class PointPos : int {
  * ordered containers.
  *
  */
-class GeoElementId
+class SketcherExport GeoElementId
 {
 public:
     /** @brief default constructor initialises object to an undefined (invalid) element.
@@ -134,9 +134,7 @@ inline constexpr GeoElementId::GeoElementId(int geoId, PointPos pos): GeoId(geoI
 {
 }
 
-inline constexpr const GeoElementId GeoElementId::RtPnt = GeoElementId(GeoEnum::RtPnt, PointPos::start);
-inline constexpr const GeoElementId GeoElementId::HAxis = GeoElementId(GeoEnum::HAxis, PointPos::none);
-inline constexpr const GeoElementId GeoElementId::VAxis = GeoElementId(GeoEnum::VAxis, PointPos::end);
+
 
 } // namespace Sketcher
 
