diff --git a/CMakeLists.txt b/CMakeLists.txt
index acd65efbe..7550cb3c9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,6 +46,13 @@ endif (${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
 
 # ================================================================================
 
+# stop python defining sprintf
+# better make it dependent on boost version (1.70)
+# and python version (3.7)
+# Most likely this is solved with python 3.8
+add_definitions(-DHAVE_SNPRINTF)
+
+
 # Allow developers to use Boost < 1.48
 if (NOT BOOST_MIN_VERSION)
     set(BOOST_MIN_VERSION 1.48)
