commit 00cef8b834b9476c406cef9a5f3f656d38cd354b
Author: lorenz <looooo@users.noreply.github.com>
Date:   Tue Jul 12 13:45:54 2022 +0200

    cross compiling: avoid failing test-program

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bbe9e6d..783d3f3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,6 +56,9 @@ if(FREECAD_USE_CCACHE)
     endif()
 endif()
 
+# for cross compiling: https://stackoverflow.com/a/53635241
+set(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")
+
 # Version and Package Name are set in the version.json file and propagate to all other
 # files with version information in them either via CMake or via the sync_version.py
 # script (in src/Tools/)
