diff --git a/src/Mod/Raytracing/CMakeLists.txt b/src/Mod/Raytracing/CMakeLists.txt
index 9c707dec7e..a882841265 100644
--- a/src/Mod/Raytracing/CMakeLists.txt
+++ b/src/Mod/Raytracing/CMakeLists.txt
@@ -9,6 +9,10 @@ set(Raytracing_Scripts
     RaytracingExample.py
 )
 
+if(WIN32)
+    add_definitions(-DBOOST_DYN_LINK)
+endif(WIN32)
+
 if(BUILD_GUI)
     list (APPEND Raytracing_Scripts InitGui.py)
 endif(BUILD_GUI)
