diff --git a/src/Mod/Raytracing/App/LuxTools.cpp b/src/Mod/Raytracing/App/LuxTools.cpp
index 86db2ee16f..22ddbcfd20 100644
--- a/src/Mod/Raytracing/App/LuxTools.cpp
+++ b/src/Mod/Raytracing/App/LuxTools.cpp
@@ -39,7 +39,6 @@
 #include <Base/Sequencer.h>
 #include <Base/Matrix.h>
 #include <App/ComplexGeoData.h>
-#include <boost/regex.hpp>
 
 
 #include "PovTools.h"
diff --git a/src/Mod/Raytracing/CMakeLists.txt b/src/Mod/Raytracing/CMakeLists.txt
index 9c707dec7e..9823b37878 100644
--- a/src/Mod/Raytracing/CMakeLists.txt
+++ b/src/Mod/Raytracing/CMakeLists.txt
@@ -1,4 +1,3 @@
-
 add_subdirectory(App)
 if(BUILD_GUI)
     add_subdirectory(Gui)
