diff --git a/cMake/FindSMESH.cmake b/cMake/FindSMESH.cmake
index 475e785cd6..0f2214513f 100644
--- a/cMake/FindSMESH.cmake
+++ b/cMake/FindSMESH.cmake
@@ -7,20 +7,20 @@
 #
  
 
-IF (CMAKE_COMPILER_IS_GNUCC)
-    FIND_PATH(SMESH_INCLUDE_DIR SMESH_Mesh.hxx
-    # These are default search paths, why specify them?
-    # /usr/include
-    # /usr/local/include
-    PATH_SUFFIXES smesh
-    )
-    FIND_LIBRARY(SMESH_LIBRARY SMESH
-    #  /usr/lib
-    #  /usr/local/lib
-    )
-ELSE (CMAKE_COMPILER_IS_GNUCC)
-    # Not yet implemented
-ENDIF (CMAKE_COMPILER_IS_GNUCC)
+FIND_PATH(SMESH_INCLUDE_DIR SMESH_Mesh.hxx
+# These are default search paths, why specify them?
+# /usr/include
+# /usr/local/include
+PATH_SUFFIXES smesh
+)
+FIND_LIBRARY(SMESH_LIBRARY SMESH
+#  /usr/lib
+#  /usr/local/lib
+)
+
+MESSAGE(STATUS "############")
+MESSAGE(STATUS "SMESH_LIBRARY: ${SMESH_LIBRARY}")
+MESSAGE(STATUS "############")
 
 SET(SMESH_FOUND FALSE)
 IF(SMESH_LIBRARY)
@@ -38,3 +38,7 @@ IF(SMESH_LIBRARY)
   )
 ENDIF(SMESH_LIBRARY)
 
+MESSAGE(STATUS "############")
+MESSAGE(STATUS "SMESH_FOUND: ${SMESH_FOUND}")
+MESSAGE(STATUS "SMESH_LIBRARIES: ${SMESH_LIBRARIES}")
+MESSAGE(STATUS "############")
\ No newline at end of file
